src/app/components/StabilityFund/index.tsx

```diff
@@ -0,0 +1,255 @@
+  const receivedCurrency = trade?.outputAmount.currency;
+  const isBnUSDGoingIn = sendSymbol === 'bnUSD';
+  const hasFundEnoughBalance =
+    sendAmount && maxSwapSize ? new BigNumber(sendAmount.toFixed(4)).isLessThanOrEqualTo(maxSwapSize) : 'loading';
+
+  const handleFundTransfer = () => {
+    if (!account) return;
+
+    const tokenToBeReceivedFromFund = isBnUSDGoingIn ? receivedCurrency?.wrapped.address || '' : undefined;
+    const stabilityFundAddress = addresses[NETWORK_ID].stabilityfund;
+
+    window.addEventListener('beforeunload', showMessageOnBeforeUnload);
+    if (bnJs.contractSettings.ledgerSettings.actived) {
+      changeShouldLedgerSign(true);
+    }
+
+    if (sendAmount && sendSymbol && receivedCurrency && feeAmount) {
+      const message = swapMessage(
+        new BigNumber(sendAmount?.toSignificant(2)).toFormat(2),
+        sendSymbol,
+        new BigNumber(sendAmount.subtract(feeAmount).toSignificant()).toFormat(2),
+        receivedCurrency.symbol || 'OUT',
+      );
+
+      const contract = bnJs.inject({ account }).getContract(sendAmount.currency.wrapped.address);
+
+      contract
+        .transfer(stabilityFundAddress, toDec(sendAmount), tokenToBeReceivedFromFund)
+        .then((res: any) => {
+          setShowFundSwapConfirm(false);
+          if (!isEmpty(res.result)) {
+            addTransaction(
+              { hash: res.result },
+              {
+                pending: message.pendingMessage,
+                summary: message.successMessage,
+              },
+            );
+            clearSwapInputOutput();
+          } else {
+            console.error(res);
+          }
+        })
+        .catch(e => {
+          console.error('error', e);
+        })
+        .finally(() => {
+          changeShouldLedgerSign(false);
+          window.removeEventListener('beforeunload', showMessageOnBeforeUnload);
+        });
+    }
+  };
+
+  const handleFundSwapConfirmDismiss = () => {
+    if (shouldLedgerSign) return;
+
+    setShowFundSwapConfirm(false);
+    changeShouldLedgerSign(false);
+  };
+
+  return (
+    <>
+      <Wrapper>
+        {hasFundEnoughBalance === 'loading' ? (
+          <Box paddingTop={4}>
+            <Spinner centered />
+          </Box>
+        ) : hasFundEnoughBalance && sendAmount && feeAmount ? (
+          <>
+            <HorizontalDivider text={t`Or`} />
+            <FundCTA>
+              <span onClick={() => (account ? setShowFundSwapConfirm(true) : toggleWalletModal())}>
+                {t`Use the Stability Fund to swap ${`${new BigNumber(sendAmount.toSignificant()).toFormat(
+                  2,
+                )} ${sendSymbol}`} for ${`${new BigNumber(sendAmount.subtract(feeAmount).toSignificant()).toFormat(
+                  2,
+                )} ${receivedCurrency?.symbol}`}`}
+              </span>
+
+              <QuestionHelper
+                text={t`The Stability Fund allows you to mint or burn bnUSD 1:1 for approved stablecoins.`}
```

Comment:
```suggestion
                text={t`The Stability Fund lets you mint or burn bnUSD 1:1 for approved stablecoins, minus a 0.5% fee.`}
```

I think it's better to be specific about the fee %, rather than just say "minus a small fee", but if the community can vote to adjust the fee, we'll need to add an API call so a UI update isn't required to show accurate info.

Reply:
updated